### DISCUSSION TITLE
Add cover entities

### DISCUSSION THREAD

**Discussion by cs33lm** [zachowj/hass-node-red#255] [General]
Thanks for your great contribution on this add-on and your support for the community! 

I recently moved from node-red-contrib-home-assistant-websocket to using only this companion, as I wrote my own node red message bus for my KNX, Shelly and Zigbee Smarthome utilizing nodered/discovery to create entities. My goal was to use vanilla node red as much as possible, so websocket was the way. Working like a charm for all my binary_sensor, sensors and switches. 

Are there any limitations to create cover entities as well as it is not possible (as stated in the README) to create them? Thx

**Comment by zachowj** [comment]
There shouldn't be any limitations. 

I have intentions to add all the Home Assistant entity types. It is just a matter of finding the time to do it. The first step is generalizing the payload sent to HA. Right now, only `state` and `attributes` exist and should be moved to a dynamic `payload` property depending on the entity type.

If you want to add an ad hoc version of the cover entity that functions close to the way the switch node does. I wouldn't be opposed to that, just know that its discovery/update format might change a little once the above changes go in.
